File: khalti-android/src/main/java/com/khalti/widget/PayContract.java
package com.khalti.widget;


import com.khalti.checkOut.api.Config;

interface PayContract {
    interface View {

        void setCustomButtonView();

        void setButtonStyle(int id);

        void setButtonText(String text);

        void setButtonClick();

        void openForm();

        void destroyCheckOut();

        void setPresenter(Presenter presenter);
    }

    interface Presenter {

        String checkConfig(Config config);

        void setCustomButtonView();

        void setButtonStyle(int id);

        void setButtonText(String text);

        void setButtonClick();

        void openForm();

        void destroyCheckOut();
    }
}

File: khalti-android/src/main/java/com/khalti/widget/PayContract.kt
package com.khalti.widget


import com.khalti.checkout.helper.Config

interface PayContract {
    interface View {

        fun setCustomButtonView()

        fun setButtonStyle(id: Int)

        fun setButtonText(text: String)

        fun setButtonClick()

        fun openForm()

        fun destroyCheckOut()

        fun setPresenter(presenter: Presenter)
    }

    interface Presenter {

        fun onCheckConfig(config: Config): String

        fun onSetCustomButtonView()

        fun onSetButtonStyle(id: Int)

        fun onSetButtonText(text: String)

        fun onSetButtonClick()

        fun onOpenForm()

        fun onDestroyCheckOut()
    }
}